natlib/svcd.c

```diff
@@ -185,3 +186,41 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+
+static int svcd_wcdispatch( lua_State *L )
+{
+    if (lua_gettop(L) != 3) return luaL_error(L, "Expected (pay, srcip, srcport)");
+    size_t parlen = lua_objlen(L, -1);
+    const char* pay = lua_tolstring(L, -1, &parlen);
+
+    lua_getglobal(L, "SVCD");
+    
+    lua_pushlightfunction(L, libmsgpack_mp_unpack);
+    lua_pushstring(L, pay);
+    lua_call(L, 1, 1);
+    uint16_t ivkid = (uint16_t) luaL_checkstring(L, 1);
```

Comment:
Perhaps you meant to get index -1 as a string? What you're doing converting the first argument (the payload) to a string, getting that string's memory address, converting the address to a uint16_t, and then assign it to ivkid. You should double-check if that is really what you want to do.

-men in #000000